### PR TITLE
Chores: put lint args into golangci-lint config file

### PR DESCRIPTION
**Pull request by MartinForReal**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Chores: put lint args into golangci-lint config file
**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:
golint is deprecated and replaced by revive.

**Release note**:
```
none
```

### CODE REVIEW

**Comment by MartinForReal** [comment]
<html>
<body>

/test pull-azuredisk-csi-driver-external-e2e-single-az
--


<br class="Apple-interchange-newline">
</body>
</html>

File: .golangci.yml [hunk 1]
```diff
@@ -1,13 +1,25 @@
 run:
```
Comment (andyzhangx) [inline .golangci.yml:1]: this `.golangci.yml` is actually not needed any more

File: .github/workflows/static.yaml [hunk 2]
```diff
@@ -16,4 +16,3 @@ jobs:
               uses: golangci/golangci-lint-action@v3
               with:
                   version: v1.54
-                  args: -E=gofmt,unused,ineffassign,revive,misspell,exportloopref,asciicheck,bodyclose,depguard,dogsled,durationcheck,errname,forbidigo -D=staticcheck --timeout=30m0s
```
Comment (andyzhangx) [inline .github/workflows/static.yaml:19]: why remove all these args?

Comment (MartinForReal) [reply .github/workflows/static.yaml:19]: the config has beem moved to .golangci.yml file.

File: .golangci.yml (hunk 1 above)
Comment (MartinForReal) [reply .golangci.yml:1]: the log shows that the golint is enabled but replaced with revive

**Comment by MartinForReal** [comment]
<html>
<body>

/test pull-azuredisk-csi-driver-e2e-capz-multi-az
--


<br class="Apple-interchange-newline">
</body>
</html>

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Pull request merged**